libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -189,73 +265,168 @@ export class BotConfiguration extends BotConfigurationBase {
+                        JSON.parse(json);
+                        await fsx.writeFile(folder + `/${luisService.id}.luis`, json, { encoding: 'utf8' });
+                        recipe.resources.push(<IDispatchResource>{
+                            type: service.type,
+                            id: service.id,
+                            name: service.name,
+                            serviceIds: (<IDispatchService>service).serviceIds
+                        });
                     }
-                }
-                dispatchService.serviceIds = validServices;
+                    break;
+                case ServiceTypes.Luis:
+                    {
+                        let luisService = <ILuisService>service;
+                        let command = `luis export version --appId ${luisService.appId} --authoringKey ${luisService.authoringKey} --versionId "${luisService.version}"`;
+                        if (progress) {
+                            progress(service, command, index, this.services.length);
+                        }
+                        let p = await exec(command);
+                        var json = p.stdout;
+                        // make sure it's json
+                        JSON.parse(json);
+                        await fsx.writeFile(folder + `/${luisService.id}.luis`, json, { encoding: 'utf8' });
+                        recipe.resources.push({
+                            type: service.type,
+                            id: service.id,
+                            name: service.name
+                        });
+                    }
+                    break;
+
+                case ServiceTypes.QnA:
+                    {
+                        let qnaService = <IQnAService>service;
+                        let command = `qnamaker export kb --kbId ${qnaService.kbId} --environment prod --subscriptionKey ${qnaService.subscriptionKey} --hostname ${qnaService.hostname} --endpointKey ${qnaService.endpointKey}`;
+                        if (progress) {
+                            progress(service, command, index, this.services.length);
+                        }
+                        let p = await exec(command);
+                        var json = p.stdout;
+                        // make sure it's json
+                        JSON.parse(json);
+                        await fsx.writeFile(folder + `/${qnaService.id}.qna`, json, { encoding: 'utf8' });
+                        recipe.resources.push({
+                            type: service.type,
+                            id: service.id,
+                            name: service.name
+                        });
+                    }
+                    break;
+
+                case ServiceTypes.Endpoint:
+                    if (progress) {
+                        progress(service, '', index, this.services.length);
+                    }
+                    recipe.resources.push(<IUrlResource>{
```

Comment:
TS provides cleaner ways to create and cast objects like this. The code here results in TSLint erorrs. Easier to just resolve those errors now.